invenio_oauthclient/contrib/cern.py

```diff
@@ -209,16 +211,68 @@ def account_setup(remote, token):
 
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
     user = token.remote_account.user
+    if not user:
+        raise Exception('`user` should not be None, something bad happened!')
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+
+        # update session object
+        groups_old = current_user.info['group']
+        groups_new = fetch_groups(res['Group'])
+        current_user.info['group'] = groups_new
```

Comment:
I don't know if Invenio is the right place for CERN people to interact with their CERN egroups:
(1) we could not offer more than a read access to the list of the user's e-groups (which we can do without storing them into invenio-groups)
(2) there are several other places where a person can view their groups at CERN, more strongly related with the user's CERN account.

@kaplun I see that usergroup data is heavily used in baskets, but not much in other places. Do you know other use cases? Even if the user is not logged in, why do we need more than just the name of the group to do notifications? Also, I don't think in the legacy code that we are doing any removal of users from e-groups or purging of the e-groups that have been deleted, so we are just accumulating information, which is not that nice.